assets/src/edit-story/components/library/panes/media/local/mediaPane.js

```diff
@@ -223,7 +223,7 @@ function MediaPane(props) {
               options={FILTERS}
               placement={Placement.BOTTOM_START}
             />
-            <Primary onClick={openMediaPicker}>
+            <Primary onClick={openMediaPicker} data-testid="uploadButton">
```

Comment:
I don't think there's a need for this; should be able to use `getByRole('button', { name: "Upload" })` in the test

Reply:
I don't see why adding this is a problem. We have `data-testid` everywhere.

Reply:
It's a best practice to use other matchers whenever possible, `data-testid` should be a last resort.

Here `getByRole('button', { name: "Upload" })` should work perfectly fine, so there's no need for a test ID.

Citing the [docs](https://testing-library.com/docs/dom-testing-library/api-queries#bytestid):

> In the spirit of the guiding principles, it is recommended to use this only after the other queries don't work for your use case. Using data-testid attributes do not resemble how your software is used and should be avoided if possible. That said, they are way better than querying based on DOM structure or styling css class names. Learn more about data-testids from the blog post "Making your UI tests resilient to change"

Reply:
Isn't `getByRole` part of the react testing library. I didn't think we can use that in puppeteer. I have used something else.

Reply:
You're right, I was too deep in looking at some Karma tests while writing this 🤦 

Your current solution looks great!